### PR TITLE
Fix outdated references to 3.6 and run pyupgrade

### PR DESCRIPTION
**Pull request by ichard26**
I also missed the accidental removal of the 3.11 classifier in the PR :sweat:

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -16,9 +16,8 @@ Also, you can try out _Black_ online for minimal fuss on the
 
 ## Installation
 
-_Black_ can be installed by running `pip install black`. It requires Python 3.6.2+ to
-run. If you want to format Jupyter Notebooks, install with
-`pip install 'black[jupyter]'`.
+_Black_ can be installed by running `pip install black`. It requires Python 3.7+ to run.
+If you want to format Jupyter Notebooks, install with `pip install 'black[jupyter]'`.
```
Comment (cooperlees) [inline docs/getting_started.md:20]: Wish we could remove the copy pasta somehow ...

File: src/black/comments.py [hunk 2]
```diff
@@ -270,8 +270,7 @@ def _generate_ignored_nodes_from_fmt_skip(
         while "\n" not in prev_sibling.prefix and prev_sibling.prev_sibling is not None:
             prev_sibling = prev_sibling.prev_sibling
             siblings.insert(0, prev_sibling)
-        for sibling in siblings:
-            yield sibling
+        yield from siblings
```
Comment (cooperlees) [inline src/black/comments.py:273]: Should this require typing change ?

File: src/black/concurrency.py [hunk 3]
```diff
@@ -58,12 +58,7 @@ def shutdown(loop: asyncio.AbstractEventLoop) -> None:
 
         for task in to_cancel:
             task.cancel()
-        if sys.version_info >= (3, 7):
-            loop.run_until_complete(asyncio.gather(*to_cancel, return_exceptions=True))
-        else:
-            loop.run_until_complete(
-                asyncio.gather(*to_cancel, loop=loop, return_exceptions=True)
-            )
+        loop.run_until_complete(asyncio.gather(*to_cancel, return_exceptions=True))
```
Comment (cooperlees) [inline src/black/concurrency.py:61]: Much cleaner.

**Review by cooperlees — APPROVED** [review]
All seems good to me.

File: src/black/comments.py (hunk 2 above)
Comment (JelleZijlstra) [reply src/black/comments.py:273]: No, the code is equivalent (IIRC there are subtle differences around exception handling but those don't matter here).

File: src/black/trans.py [hunk 4]
```diff
@@ -1256,7 +1256,7 @@ def do_transform(self, line: Line, string_idx: int) -> Iterator[TResult[Line]]:
 
         string_op_leaves = self._get_string_operator_leaves(LL)
         string_op_leaves_length = (
-            sum([len(str(prefix_leaf)) for prefix_leaf in string_op_leaves]) + 1
+            sum(len(str(prefix_leaf)) for prefix_leaf in string_op_leaves) + 1
```
Comment (JelleZijlstra) [inline src/black/trans.py:1259]: Not sure what this has to do with 3.6 but I'll take it.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**